### PR TITLE
Add capybara tests to previous stories

### LINKED ISSUE #33: Capybara Testing
#### As a developer I want to use capybara to test complete flows in my application.

Acceptance criteria:
- [ ] I have added a test for the "add a new image" flow.
- [ ] I have added a test for the "delete an existing image" flow.
- [ ] Any additional flows I've added to my application before this story are
  tested.

Discussion topics:
- [ ] What can you test in flow tests that you cannot test in controller tests?
- [ ] Going forward, when should we add flow tests?
- [ ] What are the tradeoffs between different drivers for capybara?
- [ ] How can Capybara help us with timing issues in our tests?
- [ ] How can we keep our tests DRY?

Reference material:
- https://github.com/jnicklas/capybara
- Tip 3 from http://blog.plataformatec.com.br/2011/12/three-tips-to-improve-the-performance-of-your-test-suite/

Dependencies:
- Image URL Validation
- Delete Image

### PR DESCRIPTION
**Pull request by HaominZhang**
Closes #33

### CODE REVIEW

**Comment by HaominZhang** [comment]
@rmacklin Updated, please take a look.

**Comment by rmacklin** [comment]
It's a bit difficult to track the progress with the commit comments gone from this PR. I've looked up individual commits and I believe my comments can be seen here: https://github.com/AppFolioOnboarding/Haomin_ImageUploader/compare/f64af14a0056df7e7d26fbd12056866c9a6ea470...0cb5c240568154f619602f8a8f0690c6af33c6dc and here: https://github.com/AppFolioOnboarding/Haomin_ImageUploader/compare/7e245beb248d3757d7c9eb31da304e79b5971c22...7c2545345fb0eebb4607d90f941548b2b73ee53a. Hopefully I didn't miss anything. I'm taking a look now with those comments in mind.

**Comment by HaominZhang** [comment]
I will also go through your comments again to make a double check. Thanks for providing me the reference link!

**Comment by HaominZhang** [comment]
I think I forgot to take the 2nd step as you mentioned in comment https://github.com/AppFolioOnboarding/Haomin_ImageUploader/compare/f64af14a0056df7e7d26fbd12056866c9a6ea470...0cb5c240568154f619602f8a8f0690c6af33c6dc#commitcomment-16017865. I will add it now.

File: test/flow_tests/pictures_crud_test.rb [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    click_on('New Image')
+
+    assert page.has_current_path?('/pictures/new')
+    assert_equal 'Add a new image', page.find('.page-title').text
+
+    click_on('Save link')
+    assert_equal 'Add a new image', page.find('.page-title').text
+    assert_equal "can't be blank", page.find('.picture_name.has-error .help-block').text
+    assert_equal "can't be blank", page.find('.picture_link.has-error .help-block').text
+    assert_equal "can't be blank", page.find('.picture_tag_list.has-error .help-block').text
```
Comment (rmacklin) [inline test/flow_tests/pictures_crud_test.rb]: There is some duplication here. We could extract a private method that retrieves the error message for a given field. (We did something similar in the session.)

File: test/flow_tests/pictures_crud_test.rb [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    assert_equal "/pictures/#{picture.id}", page.current_path
+
+    click_link('Delete')
+    page.driver.browser.switch_to.alert.accept
+    assert_equal '/', page.current_path
+    assert_equal "Picture was successfully deleted.", page.find('#notice').text
+    assert page.has_no_selector?("img[src='https://media.giphy.com/media/mNdOc0Aziv88E/giphy.gif']")
+  end
+
+  test 'edit an image link' do
```
Comment (rmacklin) [inline test/flow_tests/pictures_crud_test.rb]: This test is editing image tags, not an image link.

File: test/flow_tests/pictures_crud_test.rb (hunk 1 above)
Comment (HaominZhang) [inline test/flow_tests/pictures_crud_test.rb]: Sure, I will dry them out

File: test/flow_tests/pictures_crud_test.rb (hunk 2 above)
Comment (HaominZhang) [inline test/flow_tests/pictures_crud_test.rb]: Oops, I think I was careless when I manually merge the conflicts when doing rebase.. I will fix this.

**Comment by rmacklin** [comment]
OK, looks good other than the two comments above.

File: test/flow_tests/pictures_crud_test.rb (hunk 1 above)
Comment (HaominZhang) [inline test/flow_tests/pictures_crud_test.rb]: @rmacklin Currently I dried this out as ` assert_equal "can't be blank", retrieve_error_message('name')`
But do you think it is a good idea to even dry out a private method in which I put all three checks together?

Comment (rmacklin) [inline test/flow_tests/pictures_crud_test.rb]: We discussed this on Slack, but I'll answer here: you could, but I don't think that's necessary.

**Pull request merged**